sdk/keyvault/Azure.Security.KeyVault.Secrets/src/DeletedSecret.cs

```diff
@@ -21,24 +21,28 @@ public class DeletedSecret : Secret
 
         private string _recoveryId;
 
-        internal DeletedSecret()
+        internal DeletedSecret(SecretProperties properties = null) : base(properties)
         {
         }
 
         /// <summary>
         /// The identifier of the deleted secret. This is used to recover the secret.
         /// </summary>
-        public Uri RecoveryId => new Uri(_recoveryId);
+        public Uri RecoveryId
+        {
+            get => _recoveryId is null ? null : new Uri(_recoveryId);
+            internal set => _recoveryId = value?.ToString();
+        }
 
         /// <summary>
         /// The time when the secret was deleted, in UTC.
         /// </summary>
-        public DateTimeOffset? DeletedDate { get; private set; }
+        public DateTimeOffset? DeletedDate { get; internal set; }
```

Comment:
DeletedOn?

Reply:
Okay. I thought we discussed leaving things that ended in "Date" as-is, or maybe that was just more of a general conversation because Java was already doing a bunch of this. I'll check with Jonathan.